prusa/link/web/wifi-wizard/src/StateMachine.svelte

```diff
@@ -0,0 +1,190 @@
+<script context="module">
+    export const states = {
+        disabled: 0,
+        hotspotFlowStart: 1,
+        disconnectedOnStart: 2,
+        channelSwitch: 4,
+        channelSwitchFailed: 5,
+        ipsReported: 6,
+        hotspotHandoff: 7,
+        redirectImminent: 8,
+
+        // Non hotspot states
+        hostUnreachable: 100
+    };
+</script>
+
+
+<script>
+    import { onMount } from "svelte";
+
+    const UPDATE_INTERVAL = 250;
+
+    const PROBE_TIMEOUT = 7000;
+    const CHANNEL_SWITCH_START_TIMEOUT = 1000;
+    const CHANNEL_SWITCH_TIMEOUT = 15000;
+    const HOTSPOT_HANDOFF_TIMEOUT = 4000;
+
+    const ipAvailableSkipStates = [
+        states.hotspotFlowStart,
+        states.disconnectedOnStart,
+        states.channelSwitch,
+        states.channelSwitchFailed,
+    ]
+
+    const backToStartStates = [
+        states.disconnectedOnStart,
+        states.channelSwitch,
+        states.channelSwitchFailed,
+        states.ipsReported,
+        states.hotspotHandoff,
+    ]
+
+
+    const ipReachableSkipStates = [
+        states.hotspotFlowStart,
+        states.disconnectedOnStart,
+        states.channelSwitch,
+        states.channelSwitchFailed,
+        states.ipsReported,
+        states.hotspotHandoff,
+    ]
+
+    let onHotspot = false;
+    let hostReachable = true;
+    let lastSeenHost = window.performance.now();
+    let isConnecting = false;
+    let availableIp = false;
+    let reachableIp = false;
+
+    export let state = states.disabled;
+
+    export function newRawConnectionInfo(rawInfo) {
+        if (!rawInfo["active_connections"] == undefined
+            || !rawInfo["connection_details"])
+        {
+            return;
+        }
+        onHotspot = rawInfo["over_hotspot"];
+        availableIp = rawInfo["connection_details"].length > 0;
+        updateState()
+    }
+
+    export function newProbeResults(probeResults) {
+        hostReachable = probeResults.some((probeResult) => {
+            return probeResult.sameAsHost && probeResult.reachable;
+        });
+        if (hostReachable) {
+            lastSeenHost = window.performance.now();
+        }
+        reachableIp = probeResults.some((probeResult) => {
+            return !probeResult.sameAsHost && probeResult.reachable;
+        });
+        updateState()
+    }
+
+    export function newAps(aps) {
+        isConnecting = aps.some((ap) => {
+            return (ap.state == 1 || ap.state == 2)
+        });
```

Comment:
this simple functions can be one liner

```suggestion
        isConnecting = aps.some((ap) => (ap.state == 1 || ap.state == 2));
```